packages/rehype-mathjax/renderer.js

```diff
@@ -0,0 +1,25 @@
+const {mathjax} = require('mathjax-full/js/mathjax')
+const {TeX} = require('mathjax-full/js/input/tex')
+const {SVG} = require('mathjax-full/js/output/svg')
+const {liteAdaptor} = require('mathjax-full/js/adaptors/liteAdaptor')
+const {
+  RegisterHTMLHandler: registerHTMLHandler
+} = require('mathjax-full/js/handlers/html')
+const {AllPackages} = require('mathjax-full/js/input/tex/AllPackages')
+const adaptor = liteAdaptor()
+registerHTMLHandler(adaptor)
+const tex = new TeX({packages: AllPackages})
+const svg = new SVG({fontCache: 'none'})
+const html = mathjax.document('', {InputJax: tex, OutputJax: svg})
+
+module.exports = (math, display) => {
+  const css = adaptor.textContent(svg.styleSheet(html))
+  const pic = adaptor.outerHTML(html.convert(math, {display}))
+  return `
+    <span>${math}</span>
+    <script>
+    document.currentScript.previousElementSibling.attachShadow({mode:'open'});
+    document.currentScript.previousElementSibling.shadowRoot.innerHTML='<style>${css}</style>${pic}';
+    </script>
+  `.replace(/\s+/g, ' ')
```

Comment:
This seems a bit dangerous to me: replace all whitespace 🤔 Could it remove important whitespace too?

Reply:
I swiched to use hastscript instead of handling to raw string.